flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/DefaultExecutionGraphFactory.java

```diff
@@ -66,7 +63,6 @@ public class DefaultExecutionGraphFactory implements ExecutionGraphFactory {
     private final BlobWriter blobWriter;
     private final ShuffleMaster<?> shuffleMaster;
     private final JobMasterPartitionTracker jobMasterPartitionTracker;
-    private final Supplier<CheckpointStatsTracker> checkpointStatsTrackerFactory;
```

Comment:
Do you know why the factory pattern was used here and why we no longer need it?

Reply:
Good question: Initially, I thought it's an unproblematic change. The CachingSupplier was introduced with FLINK-24749. The goal was to make the `CheckpointStatsTracker` being re-used rather than being reinstantiated during rescale (it lived in the `ExecutionGraph` pre-FLINK-24749). 

The `CachingSupplier` allowed for lazy instantiation which instantiated a `CheckpointStatsTracker` only if needed (i.e. checkpointing is enabled). But I noticed that we have some side-effects during the CheckpointStatsTracker instantiation (in [CheckpointStatsTracker:129](https://github.com/XComp/flink/blob/fa9013415a332e606642f59786d6090a4c4e55b2/flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointStatsTracker.java#L129)). I.e. we have a functionality change with metrics being registered even if checkpointing is disabled. Good catch! I will address that one 👍